lib/include/openamp/virtio_mmio.h

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright (c) 2022 Wind River Systems, Inc.
+ * Based on Virtio PCI driver by Anthony Liguori, copyright IBM Corp. 2007
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef OPENAMP_VIRTIO_MMIO_H
+#define OPENAMP_VIRTIO_MMIO_H
+
+#include <openamp/virtqueue.h>
+#include <openamp/virtio.h>
+#include <metal/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Control registers */
+
+/* Magic value ("virt" string) - Read Only */
+#define VIRTIO_MMIO_MAGIC_VALUE		0x000
+
+#define MAGIC_VALUE ('v' | ('i' << 8) | ('r' << 16) | ('t' << 24))
```

Comment:
definitions, variables, structures and function that are exported should be prefixed by `virtio_mmio`